src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,48 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      this.removeFromAlive();
+    }
+  }
+
+  removeFromAlive() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
```

Comment:
[CODE PERFORMANCE] - The `removeFromAlive` method is inefficient because it is called each time an animal's health is checked, which could lead to multiple unnecessary iterations over the `Animal.alive` array. Instead, consider filtering the `Animal.alive` array directly in the `bite` method of the `Carnivore` class after an animal's health drops below zero.